ReplicaShardAllocatorSyncIdIT.testFullClusterRestartPerformNoopRecovery failed on master

Log:  https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+part1/3424/console
Build Scans: https://gradle-enterprise.elastic.co/s/c5muqklxtniog

REPRODUCE WITH: ./gradlew ':server:integTest' --tests "org.elasticsearch.gateway.ReplicaShardAllocatorSyncIdIT.testFullClusterRestartPerformNoopRecovery" \
  -Dtests.seed=1B893998BA12C55E \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-DZ \
  -Dtests.timezone=Asia/Famagusta \
  -Dcompiler.java=13

**Does not** reproduce for me. May be just an intermittent error.


stacktrace
```
Expected: <0>
     but: was <62>
java.lang.AssertionError: 

Expected: <0>
     but: was <62>

```

--- a/server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorSyncIdIT.java
+++ b/server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorSyncIdIT.java
@@ -88,7 +88,7 @@ public class ReplicaShardAllocatorSyncIdIT extends ESIntegTestCase {
         void syncFlush(String syncId) throws IOException {
             assertNotNull(indexWriter);
             try (ReleasableLock ignored = writeLock.acquire()) {
-                assertThat(getTranslogStats().getUncommittedOperations(), equalTo(0));
+                assertFalse(indexWriter.hasUncommittedChanges());
                 Map<String, String> userData = new HashMap<>(getLastCommittedSegmentInfos().userData);
                 userData.put(Engine.SYNC_COMMIT_ID, syncId);
                 indexWriter.setLiveCommitData(userData.entrySet());

